### PR TITLE
[SPARK-45470][SQL] Avoid paste string value of hive orc compression kind

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Currently, Hive supports ORC format with some compression codec( Please refer [ql/src/java/org/apache/hadoop/hive/ql/io/orc/CompressionKind.java](https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/orc/CompressionKind.java)).

Spark pasted many string literal of these compression codec. It is easy to make mistakes and reduce development efficiency.

### Why are the changes needed?
Avoid paste string value of hive orc compression kind

### Does this PR introduce _any_ user-facing change?
'No'.
Just update inner implementation.

### How was this patch tested?
Exists test cases.

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Review by LuciferYang — APPROVED** [review]
LGTM if test pass

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala [hunk 1]
```diff
@@ -87,7 +88,7 @@ class OrcHadoopFsRelationSuite extends HadoopFsRelationTest {
       val path = s"${dir.getCanonicalPath}/table1"
       val df = (1 to 5).map(i => (i, (i % 2).toString)).toDF("a", "b")
       df.write
-        .option("compression", "ZlIb")
+        .option("compression", CompressionKind.ZLIB.name())
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala:91]: Does this one test case-insensitivity? I wouldn't change tests, frankly speaking.

**Review by dongjoon-hyun — COMMENTED** [review]
Ur, very sorry, but I'd not do this ORC change when we do the samething still for Parquet , @beliefer . This only increases the compile depdency for `org.apache.hadoop.hive.ql.io.orc.CompressionKind`. We prefer the existing loose-coupled way.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala (hunk 1 above)
Comment (beliefer) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala:91]: I will revert this one.

**Comment by beliefer** [comment]
> Ur, very sorry, but I'd not do this ORC change when we do the samething still for Parquet , @beliefer . This only increases the compile depdency for `org.apache.hadoop.hive.ql.io.orc.CompressionKind`. We prefer the existing loose-coupled way.

All the change in the hive module already depends on `hive-exec`.

**Comment by dongjoon-hyun** [comment]
Just a question, is this required for SPARK-44114? Or, simply want to remove the magic strings?

**Comment by beliefer** [comment]
> Just a question, is this required for [SPARK-44114](https://issues.apache.org/jira/browse/SPARK-44114)? Or, simply want to remove the magic strings?

I investigated the source code of Hive-2 and Hive-3, the supported orc compression kind not changed.

**Comment by dongjoon-hyun** [comment]
Got it. Are we going to do the same clean-up for the other data sources like Parquet? Or Avro in `avro` module?

**Comment by beliefer** [comment]
> Got it. Are we going to do the same clean-up for the other data sources like Parquet? Or Avro in `avro` module?

Yes. I will do it.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @beliefer .

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 4.0.0~

**Comment by beliefer** [comment]
@dongjoon-hyun @LuciferYang @MaxGekk Thank you!

**Pull request closed without merging**